Support multiple Elk instances

gjbadros opened the pull request:
* Allow more than one Elk M1 alarm system to be integrated into a single hass instance.

* Introduces new "devices" schema at the top level, each of which has
  the prior configuration schema.

* Requires new version of elkm1, 0.7.14, that gwww and I just updated (thanks Glen!)

* Changes to the docs are in a separate PR on that repo.

## Breaking Change:

## Description:

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#9078

## Example entry for `configuration.yaml` (if applicable):
```
elkm1:
     - host: elk://192.168.0.2:2101/
     ....
     - host: elk://192.168.0.3:2101/
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New or updated dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard/__init__.py#L23

gjbadros committed:
Support multiple Elk instances

* Allow more than one Elk M1 alarm system to be integrated into a single hass instance.

* Introduces new "devices" schema at the top level, each of which has
  the prior configuration schema.

* Requires new version of elkm1, 0.7.14, that gwww and I just updated (thanks Glen!)

QUESTION: Should the "devices" section be optional to avoid breaking
old configuration files?  I chose not to do that for simplicity and
because I was following the doorbird code which requires the "devices"
section for all configurations even with only one device.

homeassistant commented:
Hi @gjbadros, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#22573).

Once you do that we will be able to review and accept this pull request.

Thanks!

gjbadros committed:
Fixed a bunch of hound-raised issues

Fixed issues raised by hound -- there was clearly
a tool I was supposed to run to get those warnings
before submitting the PR.  Sorry!

Updated REQUIREMENTS.

gjbadros committed:
Fixed whitespace and line-length mistakes

Also fixed unused prefix local variable lint warning.

gjbadros committed:
Fixed missing blank line

gjbadros committed:
Fixed more lint warnings.

Not sure if I missed these on the first pass or if the linter stopped
after a certain number of warnings or something else.

Switched logging to use %d and %s instead of string concatenation (per
lint request and because I imagine it migth be better performing
in some (oldish, I presume) implementations of python.

gjbadros committed:
Fixed typo in last commit.

rohankapoorcom commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -65,32 +67,34 @@ def _elk_value(val):
-        _host_validator,
-    )
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_DEVICES): vol.All(cv.ensure_list, [DEVICE_SCHEMA])
```
I would recommend removing the key `CONF_DEVICES` to remove the breaking change (similar to how the [ZoneMinder component is implemented](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/zoneminder/__init__.py)).

This would then look like:

```
CONFIG_SCHEMA = vol.Schema({
    DOMAIN: vol.All(cv.ensure_list, [DEVICE_SCHEMA])
}, extra=vol.ALLOW_EXTRA)
```

gjbadros commented:
I like that idea -- as I wrote before, I modeled this change after
Doorbird's multi-device support (
https://www.home-assistant.io/components/doorbird/).  hass arguably should
be consistent on whether the intervening tag is required, but I prefer to
avoid impacting all the singleton elkm1 users so I will switch to zm's
approach.

On Sun, Mar 31, 2019 at 11:44 PM Rohan Kapoor <notifications@github.com>
wrote:

> *@rohankapoorcom* requested changes on this pull request.
> ------------------------------
>
> In homeassistant/components/elkm1/__init__.py
> <https://github.com/home-assistant/home-assistant/pull/22573#discussion_r270731048>
> :
>
> > -            vol.Optional(CONF_TEMPERATURE_UNIT, default='F'):
> -                cv.temperature_unit,
> -            vol.Optional(CONF_AREA, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_COUNTER, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_KEYPAD, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_OUTPUT, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_PLC, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_SETTING, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_TASK, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_THERMOSTAT, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -            vol.Optional(CONF_ZONE, default={}): CONFIG_SCHEMA_SUBDOMAIN,
> -        },
> -        _host_validator,
> -    )
> +    DOMAIN: vol.Schema({
> +        vol.Required(CONF_DEVICES): vol.All(cv.ensure_list, [DEVICE_SCHEMA])
>
> I would recommend removing the key CONF_DEVICES to remove the breaking
> change (similar to how the ZoneMinder component is implemented
> <https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/zoneminder/__init__.py>
> ).
>
> This would then look like:
>
> CONFIG_SCHEMA = vol.Schema({
>     DOMAIN: vol.All(cv.ensure_list, [DEVICE_SCHEMA])
> }, extra=vol.ALLOW_EXTRA)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/22573#pullrequestreview-220941714>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbjzcn4GZCMctnP7QjIEyuvgdLGSoNFks5vcarQgaJpZM4cT-8->
> .
>

gjbadros committed:
Eliminate devices subsection in config schema

This eliminates the breaking change for configurations wanting a
singleton elk m1 instance (the majority of users, no doubt).  I did
not do it like this before because I was following the lead of the
doorbird component which introduced a devices: section when moving
to support multiple doorbells.  But Rohan Kapoor kindly pointed me
at the zoneminder component which sets the other (IMO) preferable
precedent. Will update the docs change shortly.

gjbadros commented:
Rohan, I don't see in the zoneminder documentation where it makes clear
that structurally the host configuration section can be repeated.  Having
the devices section means that I can document:

devices:
  description:
  required: true
  type: List

And then put the rest of the keys in the next section.  What am I missing?

Thanks,
Greg

On Mon, Apr 1, 2019 at 7:30 AM Greg Badros <badros@gmail.com> wrote:

> I like that idea -- as I wrote before, I modeled this change after
> Doorbird's multi-device support (
> https://www.home-assistant.io/components/doorbird/).  hass arguably
> should be consistent on whether the intervening tag is required, but I
> prefer to avoid impacting all the singleton elkm1 users so I will switch to
> zm's approach.
>
> On Sun, Mar 31, 2019 at 11:44 PM Rohan Kapoor <notifications@github.com>
> wrote:
>
>> *@rohankapoorcom* requested changes on this pull request.
>> ------------------------------
>>
>> In homeassistant/components/elkm1/__init__.py
>> <https://github.com/home-assistant/home-assistant/pull/22573#discussion_r270731048>
>> :
>>
>> > -            vol.Optional(CONF_TEMPERATURE_UNIT, default='F'):
>> -                cv.temperature_unit,
>> -            vol.Optional(CONF_AREA, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_COUNTER, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_KEYPAD, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_OUTPUT, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_PLC, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_SETTING, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_TASK, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_THERMOSTAT, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -            vol.Optional(CONF_ZONE, default={}): CONFIG_SCHEMA_SUBDOMAIN,
>> -        },
>> -        _host_validator,
>> -    )
>> +    DOMAIN: vol.Schema({
>> +        vol.Required(CONF_DEVICES): vol.All(cv.ensure_list, [DEVICE_SCHEMA])
>>
>> I would recommend removing the key CONF_DEVICES to remove the breaking
>> change (similar to how the ZoneMinder component is implemented
>> <https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/zoneminder/__init__.py>
>> ).
>>
>> This would then look like:
>>
>> CONFIG_SCHEMA = vol.Schema({
>>     DOMAIN: vol.All(cv.ensure_list, [DEVICE_SCHEMA])
>> }, extra=vol.ALLOW_EXTRA)
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/home-assistant/home-assistant/pull/22573#pullrequestreview-220941714>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABbjzcn4GZCMctnP7QjIEyuvgdLGSoNFks5vcarQgaJpZM4cT-8->
>> .
>>
>

rohankapoorcom commented:
I just read through the documentation again for ZM and it's definitely not obvious that multiple hosts can be supported. I originally got this approach from @MartinHjelmare (to avoid the breaking change when adding multiple host support). He may have some suggestions on how to best document this.

MartinHjelmare commented:
Please ask the question about the docs in the docs PR. The docs team will probably have a suggestion. So for we've just relied on the example I think.

MartinHjelmare commented on homeassistant/components/elkm1/alarm_control_panel.py:
```diff
@@ -38,9 +38,12 @@
+        elk = elk_data['elk']
+        entities = create_elk_entities(elk_data, elk.areas,
+                                       'area', ElkArea, [])
+        async_add_entities(entities, True)
```
Collect entities in a list first, then make a single call to `async_add_entities`.

gjbadros commented:
Sure; I've made the changes but don't have time to go through a full test
iteration now and will get to this by Thursday.  Thx!

On Mon, Apr 1, 2019 at 6:14 PM Martin Hjelmare <notifications@github.com>
wrote:

> *@MartinHjelmare* requested changes on this pull request.
> ------------------------------
>
> In homeassistant/components/elkm1/alarm_control_panel.py
> <https://github.com/home-assistant/home-assistant/pull/22573#discussion_r271103960>
> :
>
> > @@ -38,9 +38,12 @@
>      if discovery_info is None:
>          return
>
> -    elk = hass.data[ELK_DOMAIN]['elk']
> -    entities = create_elk_entities(hass, elk.areas, 'area', ElkArea, [])
> -    async_add_entities(entities, True)
> +    elk_datas = hass.data[ELK_DOMAIN]
> +    for _, elk_data in elk_datas.items():
> +        elk = elk_data['elk']
> +        entities = create_elk_entities(elk_data, elk.areas,
> +                                       'area', ElkArea, [])
> +        async_add_entities(entities, True)
>
> Collect entities in a list first, then make a single call to
> async_add_entities.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/22573#pullrequestreview-221412179>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbjzef380W-Nt0Edj5BspoFdC2LxsBnks5vcq8AgaJpZM4cT-8->
> .
>

gjbadros committed:
Call async_add_entities once for all the elk controllers.

Just move async_add_entities() outside of the loops across the elk m1
controllers, so it's called once for each platform.

gjbadros committed:
Call async_add_entities only once per platform.

Move it to after the loop, so it's called only once
per platform even when there are multiple elk m1 controllers.

gjbadros commented:
Just pushed these changes to my fork -- thanks for the suggestion.  What
are next steps?  I still need to resolve the documentation issue about
using devices: in the YAML file or not.  Any other changes or improvements
before this can be accepted?

On Tue, Apr 2, 2019 at 10:58 PM Greg Badros <badros@gmail.com> wrote:

> Sure; I've made the changes but don't have time to go through a full test
> iteration now and will get to this by Thursday.  Thx!
>
> On Mon, Apr 1, 2019 at 6:14 PM Martin Hjelmare <notifications@github.com>
> wrote:
>
>> *@MartinHjelmare* requested changes on this pull request.
>> ------------------------------
>>
>> In homeassistant/components/elkm1/alarm_control_panel.py
>> <https://github.com/home-assistant/home-assistant/pull/22573#discussion_r271103960>
>> :
>>
>> > @@ -38,9 +38,12 @@
>>      if discovery_info is None:
>>          return
>>
>> -    elk = hass.data[ELK_DOMAIN]['elk']
>> -    entities = create_elk_entities(hass, elk.areas, 'area', ElkArea, [])
>> -    async_add_entities(entities, True)
>> +    elk_datas = hass.data[ELK_DOMAIN]
>> +    for _, elk_data in elk_datas.items():
>> +        elk = elk_data['elk']
>> +        entities = create_elk_entities(elk_data, elk.areas,
>> +                                       'area', ElkArea, [])
>> +        async_add_entities(entities, True)
>>
>> Collect entities in a list first, then make a single call to
>> async_add_entities.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/home-assistant/home-assistant/pull/22573#pullrequestreview-221412179>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ABbjzef380W-Nt0Edj5BspoFdC2LxsBnks5vcq8AgaJpZM4cT-8->
>> .
>>
>

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -117,27 +121,47 @@ def _included(ranges, set_to, values):
+                return False
+
+        prefix = conf.get(CONF_PREFIX, "")
+        device = devices.get(prefix, None)
```
`None` is the default value returned if the key is missing in the dict.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -117,27 +121,47 @@ def _included(ranges, set_to, values):
+            elk = elkm1.Elk({'url': conf[CONF_HOST], 'userid':
+                             conf[CONF_USERNAME],
+                             'password': conf[CONF_PASSWORD]})
+            elk.connect()
```
Could the connect fail if credentials are incorrect? Do we need to catch that?

gjbadros replied:
Handled by the library.

MartinHjelmare replied:
We should fail set up if credentials are incorrect.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -146,24 +170,31 @@ def _included(ranges, set_to, values):
-def _create_elk_services(hass, elk):
+def _create_elk_services(hass, elks):
     def _speak_word_service(service):
+        elk = elks.get('', None)
```
Why would there be an empty string as dictionary key? That looks weird.

gjbadros replied:
It's a dictionary of the prefixes and for backward compatibility no prefix is allowed (making the key "").  Regardless, this is a bug in that it should've been looking up prefix.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -146,24 +170,31 @@ def _included(ranges, set_to, values):
     def _speak_phrase_service(service):
-        elk.panel.speak_phrase(service.data.get('number'))
+        prefix = service.data.get('prefix', "")
+        elk = elks.get(prefix, None)
```
Remove `None`.

gjbadros replied:
Done

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -146,24 +170,31 @@ def _included(ranges, set_to, values):
+        prefix = service.data.get('prefix', "")
+        elk = elks.get(prefix, None)
+        if elk is None:
+            _LOGGER.error("no elk m1 with prefix: '%s'", prefix)
```
Please start the logging message with capital letter.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -146,24 +170,31 @@ def _included(ranges, set_to, values):
 
     def _speak_phrase_service(service):
-        elk.panel.speak_phrase(service.data.get('number'))
+        prefix = service.data.get('prefix', "")
```
`prefix` doesn't look to be included in the service schema. Please add it.

MartinHjelmare replied:
The default is already set in the service schema. Use `dict[key]`.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -146,24 +170,31 @@ def _included(ranges, set_to, values):
+        if elk is None:
+            _LOGGER.error("no elk m1 with prefix: '%s'", prefix)
+        else:
+            elk.panel.speak_phrase(service.data.get('number'))
```
Number is a required item in the service schema so we can use `dict[key]`.

MartinHjelmare commented on homeassistant/components/elkm1/alarm_control_panel.py:
```diff
@@ -38,8 +38,12 @@
-    entities = create_elk_entities(hass, elk.areas, 'area', ElkArea, [])
+    elk_datas = hass.data[ELK_DOMAIN]
+    entities = []
+    for _, elk_data in elk_datas.items():
```
```py
for elk_data in elk_datas.values():
```

MartinHjelmare commented:
Please update the PR description after the change in direction regarding the config schema.

gjbadros commented:
Great, thanks for the careful read and I'll make these changes this
weekend.

On Fri, Apr 5, 2019 at 3:53 PM Martin Hjelmare <notifications@github.com>
wrote:

> Please update the PR description after the change in direction regarding
> the config schema.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/22573#issuecomment-480446182>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbjzYQNOvhqHipXnnCGQ40xDwmF9S8eks5vd9PPgaJpZM4cT-8->
> .
>

gjbadros committed:
Various improvements to be more idiomatic python + bug fixes

Thanks to Martin Hjelmare for the careful review and suggestions.
(All mistaken improvements and new bugs are my own.)

gjbadros committed:
Removed semicolon that lint caught.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -117,27 +122,47 @@ def _included(ranges, set_to, values):
+                _LOGGER.error("Config item: %s; %s", item, err)
+                return False
+
+        prefix = conf.get(CONF_PREFIX, "")
```
The default is already set in the config schema. Use `dict[key]`.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -117,27 +122,47 @@ def _included(ranges, set_to, values):
+        device = devices.get(prefix)
+        if device is not None:
+            if prefix == "":
+                _LOGGER.error("At most one elk m1 configuration may skip " +
```
We should write a custom config schema validator that checks that prefix is unique and set if more than one device is configured.

See the mysensors component for example.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -146,24 +171,37 @@ def _included(ranges, set_to, values):
+def _create_elk_services(hass, elks):
     def _speak_word_service(service):
-        elk.panel.speak_word(service.data.get('number'))
+        prefix = service.data.get('prefix', "")
```
The default is already set in the service schema.

MartinHjelmare commented on homeassistant/components/elkm1/__init__.py:
```diff
@@ -146,24 +171,37 @@ def _included(ranges, set_to, values):
+        if elk is None:
+            _LOGGER.error("No elk m1 with prefix for speak_word: '%s'",
+                          prefix)
+        else:
```
Return instead. Then we can outdent below.

MartinHjelmare commented on homeassistant/components/elkm1/climate.py:
```diff
@@ -18,9 +18,14 @@
-        hass, elk.thermostats, 'thermostat', ElkThermostat, []), True)
+    elk_datas = hass.data[ELK_DOMAIN]
+    entities = []
+    for _, elk_data in elk_datas.items():
```
Use `dict.values()`.

MartinHjelmare commented on homeassistant/components/elkm1/light.py:
```diff
@@ -12,10 +12,13 @@
-
+    elk_datas = hass.data[ELK_DOMAIN]
+    entities = []
+    for _, elk_data in elk_datas.items():
```
See above.

MartinHjelmare commented on homeassistant/components/elkm1/scene.py:
```diff
@@ -11,8 +11,12 @@
-    entities = create_elk_entities(hass, elk.tasks, 'task', ElkTask, [])
+    elk_datas = hass.data[ELK_DOMAIN]
+    entities = []
+    for _, elk_data in elk_datas.items():
```
See above.

MartinHjelmare commented on homeassistant/components/elkm1/sensor.py:
```diff
@@ -94,8 +97,9 @@ def _element_changed(self, element, changeset):
-        self.hass.data[ELK_DOMAIN]['keypads'][
-            self._element.index] = self.entity_id
+        elk_datas = self.hass.data[ELK_DOMAIN]
+        for _, elk_data in elk_datas.items():
```
See above.

MartinHjelmare commented on homeassistant/components/elkm1/sensor.py:
```diff
@@ -10,17 +10,20 @@
-        hass, elk.zones, 'zone', ElkZone, entities)
+    elk_datas = hass.data[ELK_DOMAIN]
+    entities = []
+    for _, elk_data in elk_datas.items():
```
See above.

MartinHjelmare commented on homeassistant/components/elkm1/switch.py:
```diff
@@ -11,8 +11,12 @@
-    entities = create_elk_entities(hass, elk.outputs, 'output', ElkOutput, [])
+    elk_datas = hass.data[ELK_DOMAIN]
+    entities = []
+    for _, elk_data in elk_datas.items():
```
See above.

gjbadros committed:
Idiomatic python improvements

Use dict.values() (instead of making it easier to add local looping variable
on the keys by using _, bar = ...items())

Use [] when the key is known to exist.

robbiet480 committed:
Further improve IndieAuth redirect_uri lookup failure logs (#23183)

StevenLooman committed:
Don't create connections between sensors. Fixes #22787 (#23202)

awarecan committed:
Add error handling for migration failure (#23383)

elupus committed:
Broadlink fixup unintended breakage from service refactor (#23408)

* Allow host/ipv6 address for broadlink service

This matches switch config and is a regression fix

* Restore padding of packets for broadlink

* Drop unused import

* Fix comment on test

andrewsayre committed:
Fix supported features gates in media_player volume up/down services (#23419)

* Correct media player feature gates

* Fix failing test

* Lint...

andrewsayre committed:
Add missing feature support flag (#23417)

cgtobi committed:
Refactor netatmo to use hass.data (#23429)

* Refactor NETATMO_AUTH to use hass.data

* Minor cleanup

* Rename conf to auth and other suggestions by Martin

* Revert webhook name change

* Rename constant

* Move auth

* Don't use hass.data.get()

* Fix auth string

nudded committed:
Fix Flux component (#23431)

* Fix Flux component

* Update manifest.json

* Update manifest.json

fredrike committed:
Fix daikin setup (#23440)

Fix daikin setup

fredrike committed:
Fix point setup (#23441)

Fix point setup

balloob committed:
Make setup more robust (#23414)

* Make setup more robust

* Fix typing

pvizeli committed:
Bump version 0.91.1

pvizeli committed:
Merge pull request #23448 from home-assistant/rc

0.92.1

zxdavb committed:
update geniushub client library to fix issue #23444 (#23450)

zxdavb committed:
update geniushub client library to fix issue #23444 (#23450)

ViViDboarder committed:
Add new nextbus sensor (#20197)

* Added new nextbus sensor

* Fix order in requirements_all after merge

* Add more flexible parsing of JSON lists

* Undo tox change

bachya committed:
Remove historical allergen and asthma sensors from IQVIA (#23258)

* Remove historical sensors from IQVIA

* Removed old imports

* Removed "Yesterday" sensors

amelchio committed:
Return of travis (#23409)

* Return .travis.yml

This reverts commit a5b03541e90a7ff7aff9f4b26573879c9c1de3fa.

* Remove coveralls

* Remove deploy

* Support only the extreme Python versions

* Ignore bleeding edge

* Remove docker service

bachya committed:
Fix broken forecast trend attribute in IQVIA (#23454)

cgtobi committed:
Fix netatmo_public sensor to use netatmo authentication (#23455)

cgtobi committed:
Fix netatmo_public sensor to use netatmo authentication (#23455)

pnbruckner committed:
Bump amcrest to 1.4.0 and use new storage_all method (#23446)

Improve sdcard sensor so it only sends one command to camera per update as opposed to the four it used to send to reduce network traffic and make data consistent. Also better handle returned values of 'unknown'.

anrudolph committed:
Added option to use self-signed certificates (#23139)

* Added option to use self-signed certificates

I defined a new option for configuration.yaml, 'verify_ssl', which is set to 'True' by default for obvious security reasons. However, in order to work with self-signed certificates, it is now possible to disable the certificate validation.(eg, I use a nextcloud instance with self-signed certificates)
Credit for code in line 57 goes to user 'gen2' on the homeassistant community, who suggested this solution. 
https://community.home-assistant.io/t/caldav-configuration/38198/25
I only took it a step further and made it a config option.

* Update calendar.py

* Added the import of CONF_VERIFY_SSL

I hope this passes the test now...

* Update homeassistant/components/caldav/calendar.py

Cool! Didn't know about that possibility, my coding experience is literally two weeks. Thanks for sharing!

Co-Authored-By: anrudolph <49680492+anrudolph@users.noreply.github.com>

* Removed some lines of code

I think, in order for the last commit to work, these lines have to be removed. Correct?

* Trying to get this passing the checks

Trying around to get the simplified code to work

amelchio committed:
No longer promote imports inside methods (#23471)

emontnemery committed:
Bump pychromecast (#23463)

emontnemery committed:
Bump pychromecast (#23463)

amelchio committed:
Improve handling of unavailable Sonos speakers (#23472)

andrewsayre committed:
Use remote image for HEOS (#23420)

balloob committed:
Fix Hue sensors returning None value (#23478)

balloob committed:
Fix Hue sensors returning None value (#23478)

zxdavb committed:
Add a warning that honeywell/EU is to be deprecated (#23469)

* Add a warning that honeywell/EU is to be deprecated

* improve warning

balloob committed:
Updated frontend to 20190427.0

balloob committed:
Updated frontend to 20190427.0

balloob committed:
Only include agent user ID in SYNC responses (#23497)

balloob committed:
Only include agent user ID in SYNC responses (#23497)

amelchio committed:
Skip flaky stream tests (#23493)

* Skip flaky test_hls tests

* Skip flaky test_recorder tests

chmielowiec committed:
Display person component as occupancy sensor HomeKit (#23451)

robmarkcole committed:
Fixes local_file camera service (#23479)

* Fixes service

Fixes service so only the target camera is updated

* Update test_camera.py

* fix lint

arsaboo committed:
Update aiolifx_effects to 0.2.2 (#23473)

amelchio committed:
Catch RequestException in influxdb writer (#23508)

atomic7777 committed:
Fixed bug with max_result (#23507)

The config option 'max_result' was assigned to not existing 'max_result' variable, it should be assigned to 'maxResult'.
The current version causes an error when max_result option is used.

emontnemery committed:
Fix cleanup of dynamic group (#23475)

emontnemery committed:
Fix cleanup of dynamic group (#23475)

balloob committed:
Return state when changing optimistic covers (#23498)

elupus committed:
Update Philips js to v0.0.8 (#23462)

* Don't send volume on volume set

This needs updated lib

* Bump version and avoid getting channels explicitly

* Add myself as codeowner

* Use version 0.0.8 instead which doesn't reuse session

Some TV's doesn't play nice with HTTP/1.1 connection
re-use.

* Add new requirement to requirements_all.txt

mcc05 committed:
Fixed AlexaPowerController to report power state for thermostats (#23468)

Fixed AlexaPowerController to report power state for thermostats, to look if state is OFF return OFF, otherwise report ON as thermostats have multiple values for ON

amelchio committed:
Continuous discovery of Sonos speakers (#23484)

amelchio committed:
Ignore stale directories (#23464)

* Ignore stale directories

* Remove redundant tests

* Revert "Remove redundant tests"

* Print warning when skipping directories

* Suggest to remove stale directory

amelchio committed:
Do not log tracebacks for influxdb write errors (#23522)

* Do not log influxdb write exceptions

* Log exception name

Swamp-Ig committed:
Allow device_class for template covers

* #23486. Allows device_class to be set for template covers

dmulcahey committed:
limit concurrent updates to avoid network flood (#23534)

bachya committed:
Add current disease data to IQVIA (#23052)

* Add current and historical disease data to IQVIA

* Added fetcher

* Added disease sensor mapping

* Changed incorrect key

* Removed other extraneous const usage

dmulcahey committed:
Allow direct binding via ZHA for the ZLL profile (#23536)

* allow binding for zll profile
* update check - review comment

emontnemery committed:
Improve logging of exceptions in async_create_task (#22689)

* Improve logging of exceptions in async_create_task

* Move wrapping+logging to util.logging

* Minor refactor, fix typing

* Add test

* Remove useless @wraps, fix confusing parameter name

* Review comment

balloob committed:
Install requirements when checking config (#23500)

* Install requirements when checking config

* PyLint

pvizeli committed:
Fix media_player alexa power control bug (#23537)

pvizeli committed:
Fix media_player alexa power control bug (#23537)

emontnemery committed:
Always print invalid configuration data (#21972)

* Always print invalide configuration data

* Print offending data as yaml

* Revert "Print offending data as yaml"

This reverts commit 01721a21a9ff918ed2c8595151ebfe55eb2f7d36.

* Do not print sensitive data

* Print MQTT topic

* Add line break

* Review comments

* review comments

dmulcahey committed:
Don't use zigpy profiles for ZHA entities (#22844)

* don't use zigpy profiles
* use sets as they're the correct structure to use

adrum committed:
Add HomeKit low battery threshold config (#23363)

etheralm committed:
Add sensor support for dyson 2018 models (#22578)

fix check for already created entities

remove hepa and carbon filter

add AQI attribute

initial commit

fix check for already created entities

remove hepa and carbon filter

add AQI attribute

add air quality component tests

fix method call tests

fix line lengths

fix pylint issues

fix docstrings

revert fan related changes

remove whitespace change

fix fan update state test

add for loop for platform initialization

add requested changes to aiq platform

change string concatenation to new style string formatting

update air quality tests

update air quality tests

refactor sensor component changes

fix pylint issues

fix debug string in the air quality component

replace failing tests for older devices

fix line length fan tests

remove dependencies const and move imports

move back imports to methods

remove whitespace from blank line

klaasnicolaas committed:
Fix problem with using Traccar event (#23543)

klaasnicolaas committed:
Fix problem with using Traccar event (#23543)

Jc2k committed:
Bump homekit_python to 0.14.0 (#23562)

amelchio committed:
Upgrade pysonos to 0.0.12 (#23560)

dmulcahey committed:
Clean up ZHA discovery logic (#23563)

* use domain constants from HA

* cleanup endpoint processing in discovery

* Whitespace.

emontnemery committed:
Migrate tests to pytest (#23544)

* Migrate tests to pytest

* Fixup

* Use loop fixture in test_check_config

* Lint

dmulcahey committed:
bump zha-quirks version (#23568)

andrewsayre committed:
Add core APIs to migrate device identifiers and entity unique_id (#23481)

* Add device identifiers migration

* Add entity unique_id migration

* Update per arch issue

* Move to existing update methods

TheLastProject committed:
Essent sensor (#23513)

* Initial commit for Essent

* Cleanup Essent component

* Update CODEOWNERS

* Move stuff to PyEssent

* Update requirements_all

* Fix PyEssent

* Move meter list to PyEssent library

* Update requirements_all

* Only check for updates once an hour

* Use PyEssent 0.10

* Fixing up Essent component

* Fix crash

* Don't add unused meter/tariff combos

* Fix lint

* Get tariffs per meter

* Don't hammer Essent API

* Fix linting errors

* Fix old description

* Fix old call

* Cleanup Essent component

cyrosy committed:
Upgrade discord.py to v1.0.1 (#23523)

mikeage committed:
Always pad mac addresses from nmap tracker with leading zeros (#23492)

MartinHjelmare committed:
Update mysensors sensor icons (#23491)

* Add some icons.
* Sort sensor types according to mysensors serial api numbering.

JonasPed committed:
Add bypass and automatic bypass switch for Danfor Air. (#23572)

Add bypass and automatic bypass switch for Danfoss Air

Swamp-Ig committed:
Ptvsd debugger component. (#23336)

* ptvsd debugger component.

* Add test case

* ptvsd as test dependency

* Fix for 3.5

* Fixed bootstrap test

* Use dict direct lookup.

* Don't need to load dependencies.

* Get the test working.

* 3.5 fix

* Set mock return value

* Put tests back, but skip them

* Change log level

Jc2k committed:
Fix min/max temp and humidity for homekit_controller climate (#23421)

* Fix min/max temp and humidity for homekit_controller climate.

* Fix typo

etheralm committed:
Fix failing state update tests (#23575)

Swamp-Ig committed:
render_with_collect method for template (#23283)

* Make entity_filter be a modifiable builder

* Add render_with_collect method

* Use sync render_with_collect and non-class based test case

* Refactor: Template renders to RenderInfo

* Freeze with exception too

* Finish merging test changes

* Removed unused sync interface

* Final bits of the diff

fabaff committed:
Upgrade youtube_dl to 2019.04.30 (#23579)

pvizeli committed:
Set up CI with Azure Pipelines

[skip ci]

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

Commit:
Bumpped the pyeconet version (#23578)

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

scop committed:
Simplify Travis cache config (#23592)

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

cdce8p committed:
Fix low_battery_threshold issue HomeKit (#23593)

Commit:
Redesign AlarmControlPanel for Homematic IP (#23565)

* redesign AlarmControlPanel

* Fix lint

* Fixes after review

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

Danielhiversen committed:
Ambiclimate (#22827)

* Ambiclimate

* Ambiclimate

* style

* Add config flow to ambicliamte

* Add config flow to ambicliamte

* update requirements_all.txt

* ambiclimate

* tests

* typo

* ambiclimate

* coverage

* add manifest.json

* services

* codeowner

* req

* ambicliamte

* style

* ambicliamte

* add to requirements all tests

* add to requirements all tests

* .coveragerc

* Add tests

* add doc

* style

* fix test

* update tests

* update tests

* update tests

* update tests

* update tests

* tests

* tests

* fix comment

ehendrix23 committed:
Fix authentication issue (#23600)

Update to pymy1 version 1.2.1 to fix authentication issue

ehendrix23 committed:
Fix authentication issue (#23600)

Update to pymy1 version 1.2.1 to fix authentication issue

cyrosy committed:
Move I/O to executor thread pool (#23589)

* Move I/O to executor thread pool

* Check if image is in whitelist_external_dir

* Move import to top of file

* Fix bad indentation

Commit:
Update bomradarloop to v0.1.3 (#23599)

teliov committed:
updated manifest.json with correct version of hangup (#23596)

Jc2k committed:
Add and improve Homekit controller pairing messages and errors (#23532)

* Be clear about pairing code format

* Handle more homekit errors while pairing

* Update en translation

* Fix log message feedback

4lloyd committed:
Added message to data_template at notify REST (#23108)

* Added message to data_template at notify rest

* Added missing newlines

bachya committed:
Fixed unhandled exception in OpenUV data update (#23611)

Jc2k committed:
Support STATE_AUTO in homekit_controller climate (#23583)

robbiet480 committed:
Fix unexpected error thrown if instance_domain is not set (#23615)

robbiet480 committed:
Fix unexpected error thrown if instance_domain is not set (#23615)

andrewsayre committed:
Bump pyheos (#23616)

andrewsayre committed:
Bump pyheos (#23616)

balloob committed:
Bumped version to 0.92.2

bachya committed:
Add support for flow sensor metrics in RainMachine (#23221)

* Initial commit

* In the clear

MartinHjelmare committed:
Clean caldav calendar tests (#23609)

emontnemery committed:
Improve sun automation tests (#23588)

* Improve sun automation tests

* Lint

* Restore timezone after test

bwarden committed:
Update pyvesync_v2 version to 0.9.7 (#23603)

pvizeli committed:
Merge pull request #23620 from home-assistant/rc

0.92.2

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

brange committed:
Prevent turning on tellduslive lights with 0 brightness (#23135)

* Prevent turning on tellduslive lights with 0 brightness

It was possible for the light compontent to get in a state where the last_brightness was set to zero, and when turning on (by calling turn_on) it would call the TellStick with 0 dim level, ergo not turning on the light.

With this, it will fall back on a 75% dim level if the level is set to zero.

* Wrap long log line over two lines

And using a more proper way of formatting the message

* Fallback to 100% instead of 75%

fabaff committed:
Update docstrings (#23624)

fabaff committed:
Order imports (#23623)

fabaff committed:
Polling is default (#23622)

Danielhiversen committed:
Updgrade Tibber library (#23630)

* update tibber library

* update tibber library

Jc2k committed:
Loosen discovery config validation to avoid breaking changes (#23625)

* Allow optional service handlers to be promoted to config flow without a breaking change

* Updated wording

* Remove period

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

scop committed:
Clean up redundant same-package ..package imports (#23587)

scop committed:
Upgrade pytest to 4.4.1 (#23584)

scop committed:
Upgrade mypy (#23586)

* Upgrade mypy to 0.701

* Enable strict equality checks

* Strict equality error fixes

* StateMachine.is_state docstring fix

max-te committed:
Allow removing Telegram reply keyboard (#23467)

*  Allow removing telegram reply keyboard by setting `keyboard` to an empty list

* Telegram keyboard: [] clearing in services.yaml

balloob committed:
Convert frontend to do client-side modern JS detection (#23618)

* Convert frontend to do client-side async pick

* Further cleanup

* Updated frontend to 20190502.0

* Fix template caching

* Remove es5 test

* Lint

* Update description

Swamp-Ig committed:
Further patch to fix of #22890 (#23627)

zxdavb committed:
Refactor evohome to prepare for water_heater (#23489)

* refactor - add const.py, change order of propertys, methods

* import client at top of file

* remove debug line

* de-lint

* delint

* add me as CODEOWNER

* remove lint hint

* delint

andrewsayre committed:
Extend play_media support (#23580)

Danielhiversen committed:
update switchbot library (#23643)

phdelodder committed:
Add "soc-thermal 1" as CPU Temp label to Glances (#23635)

Kane610 committed:
deCONZ - Retry on BridgeBusy errors (#23436)

ThaStealth committed:
Added option to select photoblack cartridge (#23433)

* Added option to select photoblack cartridge

* Update sensor.py

* Name change

GeertvanHorrik committed:
Fix bad request for some IP ONVIF camera (#22972)

* Onvif camera improvements using zeep

* Fix static code checks

* Make obtain_input_uri async

* Convert several methods to async

* Fix static checks

* Fix static checks

* Fix requirements_all.txt

* Lint improvements

* Async services

* Use onvif-zeep-async and check if PTZ service is available before creating it

* Remove some hacks that are now defined in onvif-zeep-async

* Don't log input, it might contain sensitive information

* Static code analysis fixes

* Run requirements stuff

* Fix

* Remove suds requirement

* Onvif camera improvements using zeep

* Fix static code checks

* Make obtain_input_uri async

* Convert several methods to async

* Fix static checks

* Fix static checks

* Fix requirements_all.txt

* Lint improvements

* Async services

* Use onvif-zeep-async and check if PTZ service is available before creating it

* Remove some hacks that are now defined in onvif-zeep-async

* Don't log input, it might contain sensitive information

* Static code analysis fixes

* Run requirements stuff

* Fix

* Remove suds requirement

* Use dt_util.utcnow

* Platform setup should not have a return value

* Remove explicit dependency to zeep[async]

* Bump onvif-zeep-async to 0.1.2

* Update requirements_all.txt

* Add exception handling

* Fix static checks

* Don't catch generic exceptions

* Update camera.py

SNoof85 committed:
Bump pyteleloisirs version (#23661)

* Bump pyteleloisirs version

* requirements update

fabaff committed:
Upgrade psutil to 5.6.2 (#23671)

fabaff committed:
Upgrade shodan to 1.13.0 (#23675)

fabaff committed:
Upgrade slacker to 0.13.0 (#23672)

fabaff committed:
Upgrade tapsaff to 0.2.1 (#23676)

fabaff committed:
Upgrade sqlalchemy to 1.3.3 (#23674)

bachya committed:
Remove extraneous update call in SimpliSafe (#23680)

fabaff committed:
Upgrade pysnmp to 4.4.9 (#23677)

fabaff committed:
Upgrade spotcrime to 1.0.4 (fixes #13189) (#23679)

UgaitzEtxebarria committed:
Add Bizkaibus, Biscays (Spain) bus service component (#22934)

* Updated to the current version of dev

* Added the component to .coveragerc

* Added __init__.py and manifest.json

* Changed the manifest to comply the json format

* Changes in for complete the PEP8 Code

* Fixed the api call to use PyPI package

* Fixed API correrct call

* Fixes for complete the requirements

* Added dependencies in manifest.json

* Changed the __init__.py to complete PEP8

* Simplified the __init__.py

* Runned codeowner script

* executed gen_requirements_all.py

* Direct call for dicts and unit changed to minutes

* Fixed the optional dict call, sorry, my bad, I did not that

* Deleted unused vars

* Changed optional parameter to required

* Remove blank first line

adrum committed:
Add HomeKit Television functionality  (#22968)

MartinHjelmare committed:
Clean and fix google calendar tests (#23608)

* Clean and fix google calendar tests

* Extract test calendar constant for google test
* Rewrite google calendar tests
* Clean and fix google calendar tests
* Clean and fix google component tests
* Add google conftest
* Skip flaky google calendar test

* Fix google calendar bug

* Fix yield fixture

* Set fixture names to avoid lint warning

* Fix yield fixture

Julius2342 committed:
Update requirements for pyvlx. (#23694)

MartinHjelmare committed:
Clean up mqtt device tracker tests (#23700)

* Move mqtt device tracker test under mqtt dir

* Patch config file load and save

* Clean up

* Sort imports

* Put expected value last in assertions

MartinHjelmare committed:
Clean up device tracker tests (#23695)

* Clean up device tracker tests

* Fix async load config in device tracker tests.
* Clean up lint in device tracker tests.
* Sort device tracker tests imports.
* Patch config saving in device tracker tests.
* Rename fixture.
* Rename some tests that had long cryptic names.

* Replace calls  to device_tracker.async_setup

amelchio committed:
Move Sonos services to sonos domain (#23670)

MartinHjelmare committed:
Move owntracks device tracker test under owntracks (#23701)

fredrike committed:
Add support for SET_AWAY_MODE and TURN_ON/OFF on Daikin Climate (#23585)

* add support for on/off on daikin climate

* add support for away mode on daikin climate

Adminiuga committed:
Bump ZHA modules versions. (#23705)

* Bump ZHA modules versions.

Bump bellows-homeassistant version.
Bump zigpy-homeassistant version.
Bump zigpy-xbee-homeassistant version.

* Update requirements_*.txt

dmulcahey committed:
Add additional single input cluster entries - ZHA (#23697)

* more profiles

* remove from binary sensor profile

* remove sensor profile mapping

ktnrg45 committed:
Add media_type_app to media_player (#23666)

* Add media_type_app

* Add MEDIA_TYPE_APP

* Bump 0.7.3

* Bump 0.7.3

* Bump 0.7.3

* typo

droopanu committed:
Add a TCP timeout of 5 seconds to ffmpeg (#23617)

Add a TCP timeout of 5 seconds to ffmpeg to fix stream getting stuck when network connectivity is lost
https://github.com/home-assistant/home-assistant/issues/22741

andrewsayre committed:
Bump pyheos to 0.5.2 (#23708)

fabaff committed:
Catch thethingsnetwork TypeError (#23667)

* Catch TypeError (fixes #23097)

* Re-add return values

* Update homeassistant/components/thethingsnetwork/sensor.py

Co-Authored-By: fabaff <mail@fabian-affolter.ch>

fredrike committed:
Use local constant in Daikin for STATE_OFF (#23712)

fabaff committed:
Upgrade sendgrid to 6.0.5 (#23711)

dmulcahey committed:
bump zha-quirks (#23714)

andrewsayre committed:
Update IDs after firmware upgrade in HEOS (#23641)

* Initial work

* Update tests

scop committed:
huawei_lte: handle icons for None sensor values gracefully (#23649)

Commit:
Use the URL provided by the Wink API for subscriptions. (#23710)

* Use the URL provided by the Wink API for subscriptions.

* Changed python-wink version

Danielhiversen committed:
Upgrade switchbot , fixes #23702 (#23716)

andrewsayre committed:
Fix SmartThings Samsung Air Conditioner Support (#23706)

* Bump pysmartthings to 0.6.8

* Fix AC climate device

* Fix stale comment

fabaff committed:
Share snmp constants across all platforms (#23678)

* Share constants across all platforms

* Update .coveragerc

pszafer committed:
Add TTL support and custom headers support. (#22988)

* Add TTL custom support and custom headers support.

* fix pywebpush version

* removed whitespaces surrounding docstrings.

* fixes for tests

* priority option to data

* checking of ATTR_ENDPOINT

* change checking of target to vol.Schema

* more tests

rolfberkenbosch committed:
Added new binary sensor meteoalarm to get weather alerts in Europe (#23663)

* Added new component meteoalarm

* Update sensor.py

* Update manifest.json

* Update manifest.json

* Update manifest.json

* Added file CODEOWNERS

* Modified some code, thanks @amelchio

* removed Throttle because is not being used anymore

* Update _attributes ad _state

* some cleanup

* Update sensor.py

Change sensor to binarysensor

* Rename sensor.py to binary_sensor.py

rename the file

* Update binary_sensor.py

Removed BinarySensorDevice from class

* Update binary_sensor.py

Made a mistake with BinarySensorDevice

* Update binary_sensor.py

clean up white spaces

* Update binary_sensor.py

Fix BinarySensorDevice

* Update binary_sensor.py

cleanup the import libs

* modified __init__

* fix

* final fix, thanks @amelchio

* forgot to change the sensor.py

* correct some typo in text

* fix typos

* fix another typo

* fix typo

unixko committed:
add abbreviation for current_temperature_template (#23733)

zxdavb committed:
Add water_heater to geniushub, bump client library (#23384)

* add water_heater

* add geniushub/water_heater.py

* de-lint

* de-lint 2

* remove temperature from device state attribs

* update to latest client library

* add me as CODEOWNER

* bump client

* bump client

* delint/refactor

* Update homeassistant/components/geniushub/water_heater.py

Co-Authored-By: zxdavb <david@bonnes.me>

emontnemery committed:
Add debug prints to sun (#23598)

* Add debug prints to sun

* Review comment

andrewsayre committed:
Add HEOS sign-in/out services (#23729)

* Add HEOS sign-in/out services

* Fix typo in comment

mitchellrj committed:
Fix Hue sensor integration with multiple bridges. (#23713)

Commit:
Fix effects on HUE integration for Osram bulbs (#22789)

* Fix hue effect for others

* New suggestion

* switched implementation to remove colorloop for osram bulbs

* Check ATTR_EFFECT in kwargs

* Fix  tests

fabaff committed:
Upgrade aiodns to 2.0.0 (#23743)

bieniu committed:
Add abbreviation for position_topic (#23740)

balloob committed:
Update PR template (#23520)

* Update PR template

* Update config, PR template

zxdavb committed:
Add Intergas InComfort Lan2RF gateway (#23736)

* fixed __init__.py

* add sensors

* switch to parent-child architecture

* make binary_sensor the parent

* revert parent - to water_heater

* first working version

* working, delinted (except TODO)

* update to latest client library

* remove debug code

* delint

* tweak device_state_attributes

* tweak device_state_attrbutes

* minor tweaks

* bump client library for bugfix

* improve state attributes for pumping

* update .coverage

* bugfix - binary sensors not updating

* rename to incomfort from intouch

* fix coveragerc regression

* bump client (bugfix for /protected URL)

* bump client (bugfix for /protected URL) 2

* bump client

* remove debug code

* ready for PR

* fix regression

* use xx._boiler instead of xx._objref

* improve current_temperature and delint

* use current_operation instead of state

* refactor vol.Schema

* remove unneeded instance attribute

* remove unneeded instance attribute 2

* refactor device_state_attributes

* change 'boiler' to 'heater'

* change 'boiler' to 'heater' 2

* correct a typo

* bugfix: add missing comma

* small tidy-up

StevenLooman committed:
Fix upnp logger name (#23724)

* Fix upnp logger name

* Fix more loggers

balloob committed:
Add update events to registries (#23746)

* Add update events to registries

* Add to websocket

balloob committed:
Add integration step to onboarding (#23732)

* Return an extra oauth2 auth code during onboarding

* Areas in const

* Add integration step

* Lint

* Fix tests

* Fix test

* Verify integration added to done

* Verify step is marked as done

balloob committed:
Updated frontend to 20190507.0

cgtobi committed:
Merge netatmo_public sensor into the netatmo integration (#23531)

* Merge netatmo public into netatmo integration

* Remove netatmo_public platform

* Remove dev log messages

* Improve error handling

* Check config for unsupported conditions

* Fix linter

* Reduce nested blocks

fabaff committed:
Upgrade numpy to 1.16.3 (#23673)

fabaff committed:
Upgrade Mastodon.py to 1.4.0 (#23753)

carstenschroeder committed:
New cover platform for ADS integration (#23377)

* new cover platform

* improve logging II

* remove dependencies

* fix comment

* fix review findings

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

Swamp-Ig committed:
Prompt for pin only on open / unlock (#23751)

* Prompt for pin only on open / unlock

* Fixed test cases

scop committed:
Upgrade defusedxml to 0.6.0 (#23651)

MartinHjelmare committed:
Refactor child validation (#23482)

* Try to make the process more readable and paritioned.
* Validate child values using set message.
* Only validate using relevant schemas.
* Extract node validation.
* Rework const types and schemas.
* Rework child validator.
* Enhance warning logging message.

zanglang committed:
Update Sesame component to use Candy House's library using the V3 API (#23621)

* Update Sesame component to use Candy House's library using the V3 API

* Updated requirements_all.txt

* Fix pylint warning

* Revert back to ATTR_DEVICE_ID

BKPepe committed:
Upgrade zeroconf to 0.22.0 (#23524)

balloob committed:
No longer rely on requests (#23685)

* No longer rely on requests

* Lint

* Missed a few parts

* Fix types

* Fix more types

* Update __main__.py

* Fix tests

* Lint

* Fix script

pvizeli committed:
Update azure-pipelines.yml for Azure Pipelines

balloob committed:
Updated frontend to 20190508.0

balloob committed:
Update translations

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Version bump to 0.94.0.dev0

cgtobi committed:
Bump pyatmo to v1.11 (#23766)

Commit:
Add Presence Detector Indoor to Homematic IP (#23755)

* Add presence detector indoor

use device classes constants

* Add illuminance

* isort

ties committed:
Split up yaml loaders into multiple files (#23774)

* Start moving parts of yaml utils to own module

Move parts of yaml loader out of the single large file and start
to create the structure of the yaml loaders in Ansible [0].

[0]: https://github.com/ansible/ansible/tree/devel/lib/ansible/parsing/yaml

* Finish yaml migration, update tests and mocks

  * Move code around to finish the migration
  * Update the mocks so that `open` is patched in
    `homeassistant.util.yaml.loader` instead of
    `homeassistant.util.yaml`.
  * Updated mypy ignores
  * Updated external API of `homeasistant.util.yaml`, see below:

Checked what part of the api of `homeassistant.util.yaml` was actually
called from outside the tests and added an `__ALL__` that contains only
these elements.

Updated the tests so that references to internal parts of the API (e.g.
the yaml module imported into `homeassistant.util.yaml.loader`) are
referenced directly from `homeassistant.util.yaml.loader`.

In `tests/test_yaml.py` the import `yaml` refers to
`homeassistant.util.yaml` and `yaml_loader` refers to `~.loader`.

Future work that remains for the next iteration is to create a custom
SafeConstructor and refers to that instead of monkey patching `yaml` with
custom loaders.

* Update mocks in yaml dumper, check_config

bachya committed:
Add config entry for IQVIA (#23765)

* Add config entry for IQVIA

* Updated tests and requirements

* Removed unnecessary dependency

* Fixed tests

* Reverted unintended change

elupus committed:
Add stepped volume to demo (#23759)

* Add stepped volume to demo

* Simplify somewhat to avoid extra check

sander76 committed:
Add battery binary sensor to homematic (#23067)

* first proposal

* parameter rename

* retrigger CI

* remove separate binary sensor

* remove batter_sensor

* battery device distinction at binary sensor discovery

balloob committed:
Updated frontend to 20190509.0

pszafer committed:
fix two times creating JWT headers. (#23777)

StevenLooman committed:
Sort discovered entries by 'st' to ensure getting the same device each discovery (#23763)

hunterjm committed:
Beta Fix: ONVIF (#23787)

* bump package to include wsdl

* update requirements all

dreed47 committed:
Fix for issue #23739.  Added unique_id property so (#23769)

that entities will always get mapped to the same
property ZPID code.

balloob committed:
Update sensor.py

foxel committed:
Bumped keenetic NDMS2 client version (#23786)

balloob committed:
Add support for an external step in config flow (#23782)

* Add support for an external step in config flow

* Types

* Lint

zxdavb committed:
Centralize geniushub updates (#23764)

* add hub/parent/manager

* add hub/parent/manager 2

* working now

* delint

* add back water heater

* make water_heater a child

* make water_heater a child - delint

* make water_heater a child - delint 2

* improve turn_on logic, and small tidy-up

* improve turn_on logic 2

* improve turn_on logic 3 - better docstring

* improve turn_on logic 3 - better docstring

* remove unnecessary DICT.get()s

* remove unnecessary DICT.get()s 2

* code tidy-up

* de-lint

* refactor for GeniusData

* refactor for GeniusData 2

* code tidy-up

* add missing should_poll = False

balloob committed:
Move tests to right folder (#23790)

* Move tests to right folder

* Fix test leaving files behind

amelchio committed:
Synchronize Sonos service calls (#23791)

balloob committed:
Updated frontend to 20190510.0

Adminiuga committed:
Do not add coordinator to the ZHA entities. (#23803)

hunterjm committed:
fix onvif wsdl import - take 2 (#23807)

alengwenus committed:
Add LCN climate platform (#22542)

* Add LCN climate component

* Updates of ha_state are done async

* Changes due to manifest.json

stbenjam committed:
Bump venstarcolortouch to v0.7 (#23806)

fabaff committed:
Upgrade youtube_dl to 2019.05.11 (#23808)

mvn23 committed:
Bump pyotgw to 0.4b4, fix Opentherm Gateway name in manifest.json (#23810)

balloob committed:
Fix patching right import (#23816)

zxdavb committed:
Add incomfort climate and bump client (#23830)

* Initial commit

* bump client for bugfix

* bump client for bugfix 2

* de-lint

akloeckner committed:
Make broadlink switch restore its state (#23829)

* Make broadlink switch restore its state

Method copied from pilight switch

* style

gjbadros committed:
Merge branch 'dev' of https://github.com/gjbadros/home-assistant into dev

bachya commented:
This PR is borked. You can try to rebase it and remove commits that aren't applicable, or you may need to start a new one.

gjbadros committed:
Use dict[key] instead of .get (incl. fixing typo). Use .values() instead of .items() when ignoring keys.

gjbadros commented:
Updating to the latest in the dev branch and rebasing this doesn't seem to have gotten me to a reasonable pull request, so I'll abandon this one and start a new PR after I've tested the changes against the latest (I was away from this for 5 weeks.)

gjbadros committed:
Gotta use devices.get(prefix) since we use no prefix for the singleton elk instance

gjbadros commented:
New pull request replacing this borked one is 23839.
https://github.com/home-assistant/home-assistant/pull/23839

On Sun, May 12, 2019 at 6:36 PM Aaron Bach <notifications@github.com> wrote:

> This PR is borked. You can try to rebase it and remove commits that aren't
> applicable, or you may need to start a new one.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant/pull/22573#issuecomment-491648747>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AALOHTL64HJCDEC2U6F2SSLPVDAYJANCNFSM4HCP547A>
> .
>
